src/Microsoft.AspNet.Mvc.ModelBinding/BindingInfo.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// Binding info which represents metadata associated to an action parameter.
+    /// </summary>
+    public class BindingInfo
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="ModelBinding.BindingSource"/>.
+        /// </summary>
+        public BindingSource BindingSource { get; set; }
+
+        /// <summary>
+        /// Gets or sets the binder model name.
+        /// </summary>
+        public string BinderModelName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Type"/> of the model binder used to bind the model.
+        /// </summary>
+        public Type BinderType { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the action parameter is optional.
+        /// </summary>
+        public bool? IsOptional { get; set; }
```

Comment:
What's this? we have no concept of optional parameters. If this is a webapi shim concept then it should be limited to the shim.